### ISSUE TITLE
Using '-' as separator between GroupID and ArtifactID (Maven) is a poor choice - not unique.

### ISSUE THREAD

**Issue by JWT007** [go-gitea/gitea#33036]
### Description

Copied from discussion in [[30568](https://github.com/go-gitea/gitea/issues/30568)] - originally  for a different issue and now closed with pending PR.

Gitea made a poor descision (IMHO) to use the separator '-' between groupID and artifactID.

Consider the following:

```
<groupId>foo</groupId>
<artifactId>bar-jdbc</artifactId>
<version>1-SNAPSHOT</version>

<groupId>foo-bar</groupId>
<artifactId>jdbc</artifactId>
<version>1-SNAPSHOT</version>
```

Both result in the same URL:
packages/maven/foo-bar-jdbc/1-SNAPSHOT

Although a '-' in a groupID is not "good-practice", it is not forbidden and perfectly valid and in common use.

For example:

```
<dependency>
    <groupId>commons-lang.wso2</groupId>
    <artifactId>commons-lang</artifactId>
    <version>2.6.0.wso2v1</version>
</dependency>
```

### Gitea Version

1.22.6

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

Linux

### How are you running Gitea?

-

### Database

PostgreSQL

**Comment by wxiaoguang** [comment]
Will fix it in 1.24 (could try to backport but I can't promise)

**Comment by wxiaoguang** [comment]
Now here it is  Refactor maven package registry #33049
